### PR TITLE
GEOPY-1332: adapt geoh5py for drillhole group uijson

### PR DESCRIPTION
**Pull request by MatthieuCMira**
**GEOPY-1332 - adapt geoh5py for drillhole group uijson**
Add uijson options to get values

Add test to cover the case

### CODE REVIEW

File: tests/ui_json_utils_test.py [hunk 1]
```diff
@@ -264,3 +264,17 @@ def test_is_form_test():
     assert is_form(ui_json["string_parameter"])
     ui_json["string_parameter"].pop("value")
     assert not is_form(ui_json["string_parameter"])
+
+
+def test_flatten_grou_value():
```
Comment (domfournier) [inline tests/ui_json_utils_test.py:269]:
```suggestion
def test_flatten_group_value():
```

File: geoh5py/ui_json/utils.py [hunk 2]
```diff
@@ -42,7 +42,12 @@ def flatten(ui_json: dict[str, dict]) -> dict[str, Any]:
                 if not truth(ui_json, name, "enabled"):
                     data[name] = None
                 else:
-                    data[name] = value[field]
+                    temp_value = value[field]
+
+                    if value.get("groupValue"):
+                        temp_value = (value["groupValue"], temp_value)
```
Comment (domfournier) [inline geoh5py/ui_json/utils.py:48]: Suggestion: Return a dict instead?

**Review by domfournier — COMMENTED** [review]
Looks good. We should probably add some docs about it as well under the ui.json section?

File: geoh5py/ui_json/utils.py (hunk 2 above)
Comment (MatthieuCMira) [reply geoh5py/ui_json/utils.py:48]: Agree

**Comment by MatthieuCMira** [comment]
I have a problem with the validator. 
It doesn't appear in the tests. 
Working on it.

File: geoh5py/ui_json/validation.py [hunk 3]
```diff
@@ -119,7 +119,7 @@ def _required_validators(validations):
         return val
 
     @staticmethod
-    def _validations_from_uijson(  # pylint: disable=too-many-branches
+    def _validations_from_uijson(  # pylint: disable=too-many-branches  # noqa: C901
```
Comment (domfournier) [inline geoh5py/ui_json/validation.py:122]:
```suggestion
    def _validations_from_uijson(  # pylint: disable=too-many-branches  # noqa: too complex
```

File: geoh5py/ui_json/utils.py [hunk 4]
```diff
@@ -42,7 +42,12 @@ def flatten(ui_json: dict[str, dict]) -> dict[str, Any]:
                 if not truth(ui_json, name, "enabled"):
                     data[name] = None
                 else:
-                    data[name] = value[field]
+                    temp_value = value[field]
+
+                    if value.get("groupValue"):
+                        temp_value = {"group": value["groupValue"], "data": temp_value}
```
Comment (domfournier) [inline geoh5py/ui_json/utils.py:48]:
```suggestion
                        temp_value = {"group": value["groupValue"], "value": temp_value}
```

File: tests/ui_json_utils_test.py [hunk 5]
```diff
@@ -264,3 +265,21 @@ def test_is_form_test():
     assert is_form(ui_json["string_parameter"])
     ui_json["string_parameter"].pop("value")
     assert not is_form(ui_json["string_parameter"])
+
+
+def test_flatten_group_value():
+    ui_json = deepcopy(default_ui_json)
+    ui_json["test"] = {
```
Comment (domfournier) [inline tests/ui_json_utils_test.py:272]: Can you add a template for it under `ui_json.template` like the other ones, and use it here?

File: tests/ui_json_utils_test.py [hunk 6]
```diff
@@ -264,3 +265,21 @@ def test_is_form_test():
...
+        "label": "drillhole group",
+        "value": ["test1"],
+        "groupValue": "{test}",
+        "enable": True,
+        "main": True,
+    }
+
+    validators = getattr(InputValidation, "_validations_from_uijson")(ui_json)
+    assert validators["test"]["types"] == [dict]
```
Comment (domfournier) [inline tests/ui_json_utils_test.py:281]: This could technically be None also if the parameter is optional.

**Review by domfournier — COMMENTED** [review]
Just need to add a template for completion.

**Comment by domfournier** [comment]
The template should look something like that (to be validated)
```
def drillhole_group_parameter(
    main: bool = True,
    label: str = "Property group",
    group_value: str | None = None,
    value: list[str] | None = None,
    optional: str | None = None,
) -> dict:
    """
    Dropdown menu of groups of specific types.

    :param main: Show form in main.
    :param label: Label identifier.
    :param value: Input value.
    :param group_type: Type of selectable groups.
    :param optional: Make optional if not None. Initial state provided by not None
        value.  Can be either 'enabled' or 'disabled'.

    :returns: Ui_json compliant dictionary.
    """
    form = {"main": main, "label": label, "value": value, "groupValue": group_value}

    if optional is not None:
        form.update(optional_parameter(optional))
    return form
```

File: geoh5py/ui_json/templates.py [hunk 7]
```diff
@@ -404,3 +404,47 @@ def data_value_parameter(
     if optional is not None:
         form.update(optional_parameter(optional))
     return form
+
+
+def drillhole_group_data(
+    value: list[str],
+    main: bool = True,
+    label: str = "Data channel",
+    group_type: str = "825424FB-C2C6-4FEA-9F2B-6CD00023D393",
```
Comment (domfournier) [inline geoh5py/ui_json/templates.py:413]:
```suggestion
    group_type: UUID = groups.DrillholeGroup.__TYPE_UID,
```

File: geoh5py/ui_json/templates.py [hunk 8]
```diff
@@ -404,3 +404,47 @@ def data_value_parameter(
...
+    group_value: str = "{test}",
```
Comment (domfournier) [inline geoh5py/ui_json/templates.py:414]:
```suggestion
    group_value: UUID | None = None,
```

File: geoh5py/ui_json/templates.py [hunk 9]
```diff
@@ -404,3 +404,47 @@ def data_value_parameter(
     if optional is not None:
         form.update(optional_parameter(optional))
     return form
+
+
+def drillhole_group_data(
+    value: list[str],
```
Comment (domfournier) [inline geoh5py/ui_json/templates.py:410]:
```suggestion
    value: list[str] | None = [""],
```

**Review by domfournier — COMMENTED** [review]
Check types

File: geoh5py/ui_json/templates.py (hunk 9 above)
Comment (MatthieuCMira) [reply geoh5py/ui_json/templates.py:410]: pylint doesn't like the list as a default value...

Comment (domfournier) [reply geoh5py/ui_json/templates.py]: Yea I guess it should be None by default is optional is possibe.

**Review by domfournier — APPROVED** [review]
Grand. Moving to test.

**Pull request merged**